sos/report/plugins/networking.py

```diff
@@ -141,33 +141,28 @@ def setup(self):
 
         # Get ethtool output for every device that does not exist in a
         # namespace.
-        for eth in self.listdir("/sys/class/net/"):
-            # skip 'bonding_masters' file created when loading the bonding
-            # module but the file does not correspond to a device
-            if eth == "bonding_masters":
-                continue
-            self.add_cmd_output([
-                "ethtool -%s %s" % (opt, eth) for opt in self.ethtool_shortopts
-            ])
-
-            self.add_cmd_output([
-                "ethtool " + eth,
-                "ethtool --phy-statistics " + eth,
-                "ethtool --show-priv-flags " + eth,
-                "ethtool --show-eee " + eth,
-                "tc -s filter show dev " + eth,
-                "tc -s filter show dev " + eth + " ingress",
-            ], tags=eth)
-
-            # skip EEPROM collection by default, as it might hang or
-            # negatively impact the system on some device types
-            if self.get_option("eepromdump"):
-                cmd = "ethtool -e %s" % eth
-                self._log_warn("WARNING (about to collect '%s'): collecting "
-                               "an eeprom dump is known to cause certain NIC "
-                               "drivers (e.g. bnx2x/tg3) to interrupt device "
-                               "operation" % cmd)
-                self.add_cmd_output(cmd)
+        _ecmds = ["ethtool -%s" % opt for opt in self.ethtool_shortopts]
+        self.add_device_cmd([
+            _cmd + " %(dev)s" for _cmd in _ecmds
+        ], devices='ethernet')
+
+        self.add_device_cmd([
+            "ethtool %(dev)s",
+            "ethtool --phy-statistics %(dev)s",
+            "ethtool --show-priv-flags %(dev)s",
+            "ethtool --show-eee %(dev)s",
+            "tc -s filter show dev %(dev)s",
+            "tc -s filter show dev %(dev)s ingress",
+        ], devices="ethernet")
```

Comment:
.. and the (here) missing `tags=eth` is added automatically via `_add_device_cmd default method option `tags=_dev_tags`, so no "regression" :)